components/VotePanel.tsx

```diff
@@ -12,77 +16,302 @@ import { GoverningTokenType } from '@solana/spl-governance';
 import useWalletStore, {
   EnhancedProposalState,
 } from '../stores/useWalletStore';
-import Button, { SecondaryButton } from './Button';
+import Button from './Button';
 import VoteCommentModal from './VoteCommentModal';
 import { getProgramVersionForRealm } from '@models/registry/api';
+import { VoteRegistryVoterWeight, VoterWeight } from '@models/voteWeights';
+import { BN } from '@project-serum/anchor';
+import { castVotes } from 'actions/castVotes';
+import { notify } from '@utils/notifications';
+import useVoteStakeRegistryClientStore from 'VoteStakeRegistry/stores/voteStakeRegistryClientStore';
+
+export type AccountToVoteFor = {
+  tokenRecord?: ProgramAccount<TokenOwnerRecord>;
+  voterWeight?: VoteRegistryVoterWeight | VoterWeight;
+  voteRecord?: ProgramAccount<VoteRecord>;
+  voterTokenRecord?: ProgramAccount<TokenOwnerRecord>;
+};
+
+export type AccountsToVoteFor = AccountToVoteFor[];
+
+function decideOfExecutableActions(
+  sortedAccounts: SortedAccounts,
+): {
+  showSync?: true;
+  showWithdraw?: true;
+  showVoteYes?: true;
+  showVoteNo?: true;
+} {
+  if (
+    sortedAccounts.accountsThatHaveAlreadyVoted.length > 0 &&
+    sortedAccounts.accountsReadyToVote.length > 0
+  ) {
+    return {
+      showSync: true,
+      showWithdraw: true,
+    };
+  }
+
+  if (sortedAccounts.accountsReadyToVote.length > 0) {
+    return {
+      showVoteNo: true,
+      showVoteYes: true,
+    };
+  }
+
+  if (sortedAccounts.accountsThatHaveAlreadyVoted.length > 0) {
+    return {
+      showWithdraw: true,
+    };
+  }
+
+  return {};
+}
+
+// Look at the casted votes and get the voteSide
+function getVoteSide(
+  accounts: AccountsToVoteFor,
+): YesNoVote | 'conflicted' | null {
+  return accounts.reduce((state, account) => {
+    // Ignore accounts with un-casted votes
+    if (!account.voteRecord?.account.vote) {
+      return state;
+    }
+
+    const voteSide = account.voteRecord?.account.vote?.toYesNoVote();
+
+    if (typeof voteSide === 'undefined') {
+      return state;
+    }
+
+    if (state === null) {
+      return voteSide;
+    }
+
+    if (state !== voteSide) {
+      return 'conflicted';
+    }
+
+    return voteSide;
+  }, null as YesNoVote | 'conflicted' | null);
+}
+
+type SortedAccounts = {
+  accountsReadyToVote: AccountsToVoteFor;
+  accountsThatHaveAlreadyVoted: AccountsToVoteFor;
+  accountsWithoutVotingPower: AccountsToVoteFor;
+};
+
+function sortAccountsToVoteFor(
+  accountsToVoteFor: AccountsToVoteFor,
+): SortedAccounts {
+  return accountsToVoteFor.reduce(
+    (acc, account) => {
+      // Ignore account without voterTokenRecord
+      if (!account.voterTokenRecord) {
+        return acc;
+      }
+
+      if (
+        account.voterTokenRecord.account.governingTokenDepositAmount.eq(
+          new BN(0),
```

Comment:
nit: declare in a variable outside of the component `BN_ZERO`

Reply:
<img width="249" alt="Screenshot 2022-06-17 at 11 45 58" src="https://user-images.githubusercontent.com/22965416/174251299-b61eaf66-d9cb-41a5-ab41-e8f7a78aca2b.png">

Reply:
indeed